app/src/main/java/com/udacity/shoestore/ShoeDetailFragment.kt

```diff
@@ -16,6 +17,8 @@ import timber.log.Timber
 
 class ShoeDetailFragment : Fragment() {
 
+    private val viewModel: MainViewModel by activityViewModels()
```

Comment:
Couple things here:

1. It's usual to declare the `viewModel` as a property of the `Activity` or `Fragment` class, rather than grabbing it inside a function as you used to. I'm not sure if accessing in a function is actually _broken_, but just know that the top-level property is more conventional.
2. (My bad) we needed to us `by activityViewModels()` here rather than `by viewModels()` because we explicitly wanted to be given **the same `MainViewModel` that all `Fragment`s in the `Activity` are sharing**. When we were using `by viewModels()`, we _were_ being given **a** `MainViewModel`, but not **the same** `MainViewModel` as the `ShoeListFragment` was using! That's where our disappearing shoe was going; we were updating one `MainViewModel`, but it wasn't the same `MainViewModel` as the `ShoeListFragment` was subscribed to. This is extra hard to track down because `Log`/`Timber` statements all print out as they are supposed to, and there's no hint from them that we're looking at two different `MainViewModel`s. My bad for the sloppy code suggestions in the first place here.